### PR TITLE
chore: address snyk vulnerabilities

### PR DESCRIPTION
**Pull request by Vikas26021999**
# Description

With the changes of this PR contains and ignoring vulnerabilities not applicable to us (etcd server mostly), we reached to zero:

Screenshot 2022-10-26 at 12 19 42 PM

## How

1. If we could upgrade a dependency to its vulnerable dependencies, we did so.
2. Otherwise, replace was used to force the correct version for dependencies that are not in our control

## Notion Ticket
https://www.notion.so/rudderstacks/rudder-server-snyk-vulnerabilities-56d33002b9c54385abbfed434124a43f

## Security

- [x] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

### CODE REVIEW

**Review by chrikar — COMMENTED** [review]
LGTM! Let's check with server team for this as well.

**Review by lvrach — APPROVED** [review]

**Review by chrikar — APPROVED** [review]

File: go.mod [hunk 1]
```diff
@@ -2,9 +2,20 @@ module github.com/rudderlabs/rudder-server
 
 go 1.18
 
+replace (
```
Comment (atzoum) [inline go.mod:5]: Would it be useful to add a comment here to remind us that at some point, once we upgrade the relevant libraries, these replacements can become outdated and should be removed?

**Review by atzoum — APPROVED** [review]
one minor comment with regards to replacements

File: go.mod [hunk 2]
```diff
@@ -2,9 +2,20 @@ module github.com/rudderlabs/rudder-server
...
+	github.com/aws/aws-sdk-go => github.com/aws/aws-sdk-go v1.44.123 // xitongsys/parquet-go-source uses a vulenrable version
```
Comment (chandumlg) [inline go.mod:6]:
```suggestion
	github.com/aws/aws-sdk-go => github.com/aws/aws-sdk-go v1.44.123 // xitongsys/parquet-go-source uses a vulnerable version
```

File: go.mod [hunk 3]
```diff
@@ -2,9 +2,20 @@ module github.com/rudderlabs/rudder-server
...
+	github.com/dhui/dktest => github.com/dhui/dktest v0.3.13 // many dependencies require this for testing
+	github.com/prometheus/client_golang => github.com/prometheus/client_golang v1.13.0
+	golang.org/x/crypto => golang.org/x/crypto v0.0.0-20221010152910-d6f0a8c073c2
+	golang.org/x/net => golang.org/x/net v0.0.0-20221004154528-8021a29435af
+	golang.org/x/text => golang.org/x/text v0.3.8
+	gopkg.in/yaml.v2 => gopkg.in/yaml.v2 v2.4.0 // github.com/spf13/viper uses a vulenrable version
```
Comment (chandumlg) [inline go.mod:12]:
```suggestion
	gopkg.in/yaml.v2 => gopkg.in/yaml.v2 v2.4.0 // github.com/spf13/viper uses a vulnerable version
```

File: go.mod [hunk 4]
```diff
@@ -2,9 +2,20 @@ module github.com/rudderlabs/rudder-server
...
+	gopkg.in/yaml.v3 => gopkg.in/yaml.v3 v3.0.1 // github.com/spf13/viper uses a vulenrable version
```
Comment (chandumlg) [inline go.mod:13]:
```suggestion
	gopkg.in/yaml.v3 => gopkg.in/yaml.v3 v3.0.1 // github.com/spf13/viper uses a vulnerable version
```

**Review by chandumlg — APPROVED** [review]

File: go.mod (hunk 1 above)
Comment (lvrach) [reply go.mod:5]: Yeah, eventually, we should do that, and we also need to keep the replacement versions up to date.

I am not fun of `replace` solution, and I tried to use it as little as possible. I am considering ways we can avoid it or make it easier to maintain.

**Review by fracasula — APPROVED** [review]

**Pull request merged**